reactive/kotlinx-coroutines-reactor/src/Mono.kt

```diff
@@ -133,87 +132,66 @@ public fun <T> CoroutineScope.mono(
 ): Mono<T> = monoInternal(this, context, block)
 
 /**
- * Awaits the first value from the given publisher without blocking the thread and returns the resulting value, or, if
- * the publisher has produced an error, throws the corresponding exception.
- *
- * This suspending function is cancellable.
- * If the [Job] of the current coroutine is cancelled or completed while the suspending function is waiting, this
- * function immediately cancels its [Subscription] and resumes with [CancellationException].
- *
- * @throws NoSuchElementException if the publisher does not emit any value
+ * This function is deprecated in favor of [Mono.awaitSingle].
```

Comment:
This phrasing implicitly implies that this function was non-deprecated prior to this version.
I suggest rephrasing it more explicitly, e.g.: "This is a deprecated lint function to improve user experience and guard against ambiguous Mono usages." and adding example to the end of KDoc:

```
Example of ambiguous usage:
\```
// The signature of the call suggests that one or zero values can be returned from 'findById', but 'awaitFirstOrNull' instead of 'awaitSingleOrNull' may trick the reader into believing that more than one value can be returned
myDbClient.findById(uniqueId).awaitFirstOrNull()
\```
```

Reply:
Changed. What do you think?

Reply:
👍